addons/knowledge/models/knowledge_article.py

```diff
@@ -1044,7 +1044,7 @@ def _get_partner_member_permissions(self, partner_id):
         The member model is fully flushed before running the request. """
         self.env['knowledge.article'].flush()
         self.env['knowledge.article.member'].flush()
-        args = [partner_id]
+        args = [partner_id.id]
```

Comment:
Rename partner_id to partner then maybe ? partner_id generally indicates a record :)